civiform/civiform
# Add screen to prevent duplicate application submissions

### Description

This PR prevents submission of duplicate applications and adds a screen to be shown to applicants if they attempt to re-submit an application without making any changes to it.

![image](https://github.com/civiform/civiform/assets/24319951/410ef1e0-588c-4cf5-a691-b37a94649a0a)


## Release notes

With this change, applicants will no longer be able to re-submit applications without editing them. If an applicant attempts to submit a duplicate application, they will be shown a screen letting them know there are no changes to save and asking if they would like to continue editing or exit the application.

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes

- [x] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings) (waiting on translations to merge)

### Instructions for manual testing

1. Seed program and question data
2. Log in as an admin and publish the program(s), log out
3. As an applicant (logged in or out), complete an application and submit
4. Click "Apply to another program"
5. Click "Edit" on the program you've already applied for
6. Click "Submit" without making any changes (also try opening up a question and saving again without making changes, behavior should be the same)
7. See screen telling you no changes were made

### Issue(s) this completes

Fixes #4869
